packages/client/tests/functional/too-many-bind-variables/tests.ts

```diff
@@ -0,0 +1,93 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type * as imports from './node_modules/@prisma/client'
+
+declare let prisma: imports.PrismaClient
+
+testMatrix.setupTestSuite(
+  ({ providerFlavor }) => {
+    const N_LARGE = 32768
+    const N_LARGER = 65536
+
+    function generatedIds(n: number) {
+      // ["1","2",...,"n"]
+      const ids = Array.from({ length: n }, (_, i) => i + 1).map((id) => `${id}`)
+      return ids
+    }
+
+    function selectWithIds(ids: string[]) {
+      return prisma.model.findMany({
+        where: {
+          OR: [
+            {
+              id: { in: ids },
+            },
+            {
+              id: { in: ids },
+            },
+          ],
+        },
+      })
+    }
+
+    testIf(providerFlavor === undefined)(
+      'Using Rust drivers with massive filter list results in "too many bind variables" error',
+      async () => {
+        for (const n of [N_LARGE, N_LARGER]) {
+          const ids = generatedIds(n)
+
+          try {
+            await selectWithIds(ids)
+            // unreachable
+            expect(true).toBe(false)
+          } catch (error) {
+            const e = error as Error
+
+            expect(e.message).toContain('Assertion violation on the database')
+            expect(e.message).toContain('too many bind variables in prepared statement')
+          }
+        }
+      },
+    )
+
+    testIf(providerFlavor !== undefined)(
+      'Using Driver Adapters with massive filter list does not cause errors',
+      async () => {
+        const n = N_LARGE
+        const ids = generatedIds(n)
+        await selectWithIds(ids)
+      },
+    )
```

Comment:
@miguelff, this shows Driver Adapters succeeding on queries with `32768` parameters, despite Rust Drivers failing on the same number